source/common/ssl/context_impl.cc

```diff
@@ -510,9 +520,31 @@ bssl::UniquePtr<SSL> ClientContextImpl::newSsl() const {
     SSL_set_renegotiate_mode(ssl_con.get(), ssl_renegotiate_freely);
   }
 
+  if (max_session_keys_ > 0) {
+    absl::WriterMutexLock l(&session_keys_mu_);
+    if (!session_keys_.empty()) {
+      SSL_SESSION* session = session_keys_.front().get();
```

Comment:
Can you add some comments here on why picking the front of the queue is the right thing to do? I.e. why not the third item in the Q?

Reply:
Done.